### PR TITLE
Introduction of QowaivBsonSerializerCollection

### PR DESCRIPTION
**Pull request by Corniel**
To reduce the registration hell of BSON/JSON convention based types, the `QowaivBsonSerializionProvider` is introduced.

### CODE REVIEW

**Review by Sjaaky — COMMENTED** [review]
1 typo

File: test/Qowaiv.Json.UnitTests/Bson.MongoDB/BSON_serialization_provider_sepcs.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+﻿using Qowaiv.Bson.MongoDB;
+using Qowaiv.Customization;
+using Qowaiv.Identifiers;
+using Qowaiv.Json.UnitTests.Models;
+using Specs.Models;
+
+namespace BSON.MongoDB.BSON_serialization_provider_sepcs;
```
Comment (LauraXiulan) [inline test/Qowaiv.Json.UnitTests/Bson.MongoDB/BSON_serialization_provider_sepcs.cs:7]: specs, also in filename

File: test/Qowaiv.Json.UnitTests/Bson.MongoDB/BSON_serialization_provider_sepcs.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    [Test]
+    public void SVO_with_FromJson_bool_only()
+        => provider.GetSerializer(typeof(SvoWithFromJsonBoolOnly)).Should().BeNull();
+ 
+    [Test]
+    public void DTO()
+        => provider.GetSerializer(typeof(DtoClass)).Should().BeNull();
+}
+
+class Cashes_for_performance_reasons
```
Comment (LauraXiulan) [inline test/Qowaiv.Json.UnitTests/Bson.MongoDB/BSON_serialization_provider_sepcs.cs]: Caches

**Review by LauraXiulan — APPROVED** [review]

**Review by Sjaaky — APPROVED** [review]
lgtm

**Pull request merged**